common/models/templates/src/tokenization.ts

```diff
@@ -33,6 +33,42 @@ namespace models {
     let leftSpans  = wordBreaker(context.left || '') || [];
     let rightSpans = wordBreaker(context.right || '') || [];
 
+    let leftTail: Span;
+    if(leftSpans.length > 0) {
+      leftTail = leftSpans[leftSpans.length - 1];
+    }
+
+    // Handle any directly-pre-caret 'detokenization' cases - where for this _specific_ context,
+    // we should not make a token division where one normally would exist otherwise.
+    // Refer to https://github.com/keymanapp/keyman/issues/6572.
+    if(leftSpans.length > 1 && rightSpans.length == 0) {
+      const leftTailBase = leftSpans[leftSpans.length - 2];
+
+      // if the two spans are adjacent...
+      // Nested if statements in case we need another 'detokenization' condition in the future;
+      // the outer condition should hold for all such cases.
```

Comment:
Please drop this comment -- we shouldn't generally give code-structure justification comments like this. A future developer would still need to carefully review their code changes so at best this may cause them to take less care than they should.

```suggestion
```